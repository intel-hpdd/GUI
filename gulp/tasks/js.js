//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
