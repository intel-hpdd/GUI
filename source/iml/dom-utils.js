// @flow

type QuerySelector<T: HTMLElement> = (HTMLElement | Document, string) => T;

export const querySelector: QuerySelector<*> = (
  elm: HTMLElement | Document,
  sel: string
) => {
  const selected = elm.querySelector(sel);

  if (!selected)
    throw new Error(
      `Could not find selector ${sel} for element ${elm.nodeName}`
    );

  return selected;
};
