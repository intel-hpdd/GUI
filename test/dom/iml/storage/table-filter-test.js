// @flow

import { render } from "inferno";
import { renderToSnapshot } from "../../../test-utils.js";
import { querySelector } from "../../../../source/iml/dom-utils.js";

describe("storage component", () => {
  let TableFilter, mockStore;

  beforeEach(() => {
    mockStore = { dispatch: jest.fn() };
    jest.mock("../../../../source/iml/store/get-store.js", () => mockStore);

    TableFilter = require("../../../../source/iml/storage/table-filter.js").default;
  });

  describe("TableFilter", () => {
    let state;
    beforeEach(() => {
      state = {
        sortDirection: "",
        sortKey: "name",
        storageResourceClassIndex: 0,
        storageResourcesClasses: [
          {
            class_name: "EMCPower",
            columns: [
              {
                label: "Size",
                name: "size"
              },
              {
                label: "Filesystem type",
                name: "filesystem_type"
              },
              {
                label: "Uuid",
                name: "uuid"
              }
            ],
            fields: [
              {
                class: "Bytes",
                label: "Size",
                name: "size",
                optional: false,
                user_read_only: false
              },
              {
                class: "Boolean",
                label: "Filesystem type",
                name: "filesystem_type",
                optional: true,
                user_read_only: false
              },
              {
                class: "String",
                label: "Uuid",
                name: "uuid",
                optional: false,
                user_read_only: false
              }
            ],
            id: 1,
            label: "linux-EMCPower",
            modified_at: "2017-07-19T13:58:02.615410",
            plugin_internal: true,
            plugin_name: "linux",
            resource_uri: "/api/storage_resource_class/1/",
            user_creatable: false
          },
          {
            class_name: "LinuxDeviceNode",
            columns: [
              {
                label: "Host id",
                name: "host_id"
              },
              {
                label: "Path",
                name: "path"
              },
              {
                label: "Logical drive",
                name: "logical_drive"
              }
            ],
            fields: [
              {
                class: "Integer",
                label: "Host id",
                name: "host_id",
                optional: false,
                user_read_only: false
              },
              {
                class: "PosixPath",
                label: "Path",
                name: "path",
                optional: false,
                user_read_only: false
              },
              {
                class: "ResourceReference",
                label: "Logical drive",
                name: "logical_drive",
                optional: true,
                user_read_only: false
              }
            ],
            id: 2,
            label: "linux-LinuxDeviceNode",
            modified_at: "2017-07-19T13:58:02.693359",
            plugin_internal: true,
            plugin_name: "linux",
            resource_uri: "/api/storage_resource_class/2/",
            user_creatable: false
          },
          {
            class_name: "LocalMount",
            columns: [
              {
                label: "Fstype",
                name: "fstype"
              },
              {
                label: "Mount point",
                name: "mount_point"
              }
            ],
            fields: [
              {
                class: "String",
                label: "Fstype",
                name: "fstype",
                optional: false,
                user_read_only: false
              },
              {
                class: "String",
                label: "Mount point",
                name: "mount_point",
                optional: false,
                user_read_only: false
              }
            ],
            id: 3,
            label: "linux-LocalMount",
            modified_at: "2017-07-19T13:58:02.876150",
            plugin_internal: true,
            plugin_name: "linux",
            resource_uri: "/api/storage_resource_class/3/",
            user_creatable: false
          }
        ],
        storageResources: {
          object: []
        }
      };
    });

    it("should render correctly", () => {
      expect(
        renderToSnapshot(<TableFilter classes={state.storageResourcesClasses} idx={state.storageResourceClassIndex} />)
      ).toMatchSnapshot();
    });

    it("should select a specific option", () => {
      expect.assertions(2);

      const root = document.createElement("div");
      querySelector(document, "body").appendChild(root);
      render(<TableFilter classes={state.storageResourcesClasses} idx={state.storageResourceClassIndex} />, root);

      const select: HTMLSelectElement = (querySelector(root, "select"): any);

      select.selectedIndex = 1;
      select.dispatchEvent(new Event("change"));

      expect(mockStore.dispatch).toHaveBeenCalledOnceWith({
        type: "SET_STORAGE_SELECT_INDEX",
        payload: 1
      });

      expect(mockStore.dispatch).toHaveBeenCalledOnceWith({
        payload: true,
        type: "SET_STORAGE_TABLE_LOADING"
      });
    });
  });
});
