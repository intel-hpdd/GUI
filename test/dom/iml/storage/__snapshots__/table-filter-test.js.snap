// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage component TableFilter should render correctly 1`] = `
<div>
  <div
    class="form-group well"
  >
    <label
      class="control-label"
    >
      Filter Resource Class
    </label>
    <select
      class="form-control"
    >
      <option
        value="EMCPower,linux"
      >
        linux-EMCPower
      </option>
      <option
        value="LinuxDeviceNode,linux"
      >
        linux-LinuxDeviceNode
      </option>
      <option
        value="LocalMount,linux"
      >
        linux-LocalMount
      </option>
    </select>
  </div>
</div>
`;
